.circleci/config.yml

```diff
@@ -103,8 +109,18 @@ jobs:
     parameters: *parameters
     docker: *docker
     steps:
-      - *install_testing_dependencies
       - checkout
+      - when:
+          condition:
+            equal: [ bullseye, << parameters.image >> ]
+          steps:
+            - *install_poetry_bullseye
+      - when:
+          condition:
+            equal: [ bookworm, << parameters.image >> ]
+          steps:
+            - *install_poetry_bookworm
```

Comment:
I tried to refactor this into an anchor to DRY it, but CircleCI's config parser wasn't happy with placing conditions in anchors. If you see a way to get that to work, happy to refactor.

Reply:
Makes sense, done!